File: sample/use_cocoapods/objectivec/GNAdGoogleFullscreenInterstitialAdapterSample/GNAdGoogleFullscreenInterstitialAdapterSample/ViewController.m
//
//  ViewController.m
//  GNAdGoogleFullscreenInterstitialAdapterSample
//

#import "ViewController.h"
#import "Util.h"
@import GoogleMobileAds;

@interface ViewController ()<GADFullScreenContentDelegate, UITextFieldDelegate>

@property (weak, nonatomic) IBOutlet UITextField *unitIdView;
@property (weak, nonatomic) IBOutlet UIButton *buttonLoad;
@property (weak, nonatomic) IBOutlet UIButton *buttonShow;

@property(nonatomic, strong) GAMInterstitialAd *interstitial;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _buttonShow.enabled = NO;
    _unitIdView.delegate = self;

    GADMobileAds.sharedInstance.requestConfiguration.testDeviceIdentifiers = @[ [Util admobDeviceID] ];
}

- (IBAction)downButtonLoad:(id)sender {
    NSString* unitId = _unitIdView.text;
    if (unitId.length == 0) {
        return;
    }

    GAMRequest *request = [GAMRequest request];
    [GAMInterstitialAd loadWithAdManagerAdUnitID:unitId request:request completionHandler:^(GAMInterstitialAd *ad, NSError *error) {
        self.buttonLoad.enabled = YES;
        if (error) {
            NSLog(@"ViewController: downButtonLoad error = %@", error.localizedDescription);
            return;
        }
        NSLog(@"ViewController: downButtonLoad ad loaded.");
        self.buttonShow.enabled = YES;
        self.interstitial = ad;
        self.interstitial.fullScreenContentDelegate = self;
    }];

    _buttonLoad.enabled = NO;
}

- (IBAction)downButtonShow:(id)sender {
    if (self.interstitial) {
        NSLog(@"ViewController: downButtonShow show.");
        [self.interstitial presentFromRootViewController:self];
    } else {
        NSLog(@"ViewController: downButtonShow not show.");
    }
}


#pragma mark GADFullScreenContentDelegate
- (void)adDidPresentFullScreenContent:(id)ad {
    NSLog(@"ViewController: adDidPresentFullScreenContent");
}

- (void)ad:(id)ad didFailToPresentFullScreenContentWithError:(NSError *)error {
    NSLog(@"ViewController: didFailToPresentFullScreenContentWithError error = %@", [error localizedDescription]);
}

- (void)adDidDismissFullScreenContent:(id)ad {
    NSLog(@"ViewController: adDidDismissFullScreenContent");
    _buttonShow.enabled = NO;
    _interstitial = nil;
}


#pragma mark UITextFieldDelegate
- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}


@end

File: sample/use_cocoapods/swift/GNAdGoogleFullscreenAdapterSample/GNAdGoogleFullscreenAdapterSample/ViewController.swift
//
//  ViewController.swift
//  GNAdGoogleFullscreenAdapterSample
//

import UIKit
import GoogleMobileAds

class ViewController: UIViewController {

    @IBOutlet weak var unitIdView: UITextField!
    @IBOutlet weak var buttonLoad: UIButton!
    @IBOutlet weak var buttonShow: UIButton!

    var interstitial: GAMInterstitialAd!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        unitIdView.delegate = self
        buttonShow.isEnabled = false

        GADMobileAds.sharedInstance().requestConfiguration.testDeviceIdentifiers = [Util.admobDeviceID()]
    }
    
    @IBAction func downButtonLoad(_ sender: Any) {
        let unitId = unitIdView.text!
        if unitId.isEmpty {
            return
        }

        let request = GAMRequest()
        GAMInterstitialAd.load(withAdManagerAdUnitID: unitId, request: request, completionHandler: { (ad, error) in
            self.buttonLoad.isEnabled = true
            if let error = error {
                print("ViewController: downButtonLoad error = \(error.localizedDescription)")
                return
            }
            print("ViewController: downButtonLoad ad loaded.")
            self.buttonShow.isEnabled = true
            self.interstitial = ad
            self.interstitial.fullScreenContentDelegate = self
        })

        buttonLoad.isEnabled = false
    }
    
    @IBAction func downButtonShow(_ sender: Any) {
        if let ad = interstitial {
            print("ViewController: downButtonShow show.")
            ad.present(fromRootViewController: self)
        } else {
            print("ViewController: downButtonShow not show.")
        }
    }

}
//MARK GADInterstitialDelegate
extension ViewController : GADFullScreenContentDelegate {
    func adDidPresentFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print("ViewController: adDidPresentFullScreenContent")
    }

    func ad(_ ad: GADFullScreenPresentingAd, didFailToPresentFullScreenContentWithError error: Error) {
      print("ViewController: didFailToPresentFullScreenContentWithError error = \(error.localizedDescription).")
    }

    func adDidDismissFullScreenContent(_ ad: GADFullScreenPresentingAd) {
        print("ViewController: adDidDismissFullScreenContent")
        buttonShow.isEnabled = false
        interstitial = nil
    }

}
//MARK GADInterstitialDelegate
extension ViewController : UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        // Hide the keyboard when press return key in a UITextField
        unitIdView.resignFirstResponder()
        return true;
    }

}

